### PR TITLE
added StringFinder class

### PR DESCRIPTION
**Pull request by WilleTheCoder**

### CODE REVIEW

File: StringFinder.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                if (next.contains(word)) {
+                    System.out.println(next);
+                }
+            }
+            reader.close();
+        } catch (Exception E) {
+            System.out.println("Error while reading file");
+        }
+    }
+
```
Comment (danielel1) [inline StringFinder.java:37]: why double new line?

File: StringFinder.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                System.out.println("No valid method used");
+            }
+        }
+    }
+
+    /*
+     * print lines where <pattern> is matched in <file>
+     * Example: "search cat demo.txt"
+     */
+    private static void searchFile(String word, String filename) {
```
Comment (danielel1) [inline StringFinder.java]: <pattern> = word?, <file> filename, good to use same name on both places

File: StringFinder.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+public class StringFinder {
+
+    public static void main(String[] args) {
+        if (args.length != 3) {
+            System.out.println("Wrong number of input parameters!");
+        } else {
+            if (args[0].equals("search")) {
+                searchFile(args[1], args[2]);
+            } else {
+                System.out.println("No valid method used");
```
Comment (danielel1) [inline StringFinder.java:13]: valid method?

File: StringFinder.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import java.io.File;
+import java.util.Scanner;
+
+public class StringFinder {
+
+    public static void main(String[] args) {
+        if (args.length != 3) {
+            System.out.println("Wrong number of input parameters!");
+        } else {
+            if (args[0].equals("search")) {
```
Comment (danielel1) [inline StringFinder.java:11]: what if the input is in none strings?

File: StringFinder.java (hunk 3 above)
Comment (WilleTheCoder) [reply StringFinder.java:13]: yes valid

File: StringFinder.java (hunk 2 above)
Comment (WilleTheCoder) [reply StringFinder.java]: changed parameters as defined in the comment

File: StringFinder.java (hunk 4 above)
Comment (WilleTheCoder) [reply StringFinder.java:11]: Fixed so the program can handle other types by converting them to strings

Comment (WilleTheCoder) [reply StringFinder.java:11]: For instance ints.

**Review by danielel1 — APPROVED** [review]

**Pull request merged**